addons/web/static/tests/core/debug/debug_manager_tests.js

```diff
@@ -720,4 +720,186 @@ QUnit.module("DebugMenu", (hooks) => {
             ]
         );
     });
+
+    QUnit.test("set defaults: setting default value for datetime field", async (assert) => {
+        assert.expect(3);
+
+        prepareRegistriesWithCleanup();
+        patchWithCleanup(odoo, {
+            debug: true,
+        });
+
+        registry.category("services").add("user", makeFakeUserService());
+        registry.category("debug").category("form").add("setDefaults", setDefaults);
+
+        const serverData = getActionManagerServerData();
+        serverData.actions[1234] = {
+            id: 1234,
+            xml_id: "action_1234",
+            name: "Partners",
+            res_model: "partner",
+            res_id: 1,
+            type: "ir.actions.act_window",
+            views: [[18, "form"]],
+        };
+        serverData.models.partner.fields.datetime = {string: 'Datetime', type: 'datetime'}
+        serverData.views["partner,18,form"] = `
+            <form>
+                <field name="datetime"/>
+            </form>`;
+        serverData.models["ir.ui.view"] = {
+            fields: {},
+            records: [{ id: 18 }],
+        };
+        serverData.models.partner.records = [{
+            id: 1,
+            display_name: "p1",
+            datetime: "2024-01-24 16:46:16"
+        }];
+
+        const mockRPC = async (route, args) => {
+            if (args.method === "check_access_rights") {
+                return Promise.resolve(true);
+            }
+            if (args.method === "set" && args.model === "ir.default") {
+                assert.deepEqual(args.args, ["partner", "datetime", "2024-01-24 16:46:16", true, true, false]);
+                return true;
+            }
+        };
+        const webClient = await createWebClient({ serverData, mockRPC });
+        await doAction(webClient, 1234);
+        await click(target.querySelector(".o_debug_manager button"));
+        await click(target.querySelector(".o_debug_manager .dropdown-item"));
+        assert.containsOnce(target, ".modal");
+
+        const select = target.querySelector(".modal #formview_default_fields");
+        select.value = "datetime";
+        select.dispatchEvent(new Event("change"));
+        await nextTick();
+        await click(target.querySelectorAll(".modal .modal-footer button")[1]);
+        assert.containsNone(target, ".modal");
```

Comment:
to decrease the size of the tests that are very similar, we could probably repeat this section three times in a loop:

```js
for (const fieldName of ['datetime', 'reference', 'm2o']) {
    // ...
    select.value = fieldName;
    // ...
}
```

then eg in the `mockRPC` do a `self.step([args.args[1], args.args[2]]);`  and after the loop a:

```js
assert.verifySteps([
    ["datetime", "2024-01-24 16:46:16"],
    ["reference", {"displayName": "Test", "resId": 1, "resModel": "pony"}],
    ["m2o", 1],
]);
```
(the server data and view should also be adapted to have the multiple fields)